### PR TITLE
bluetooth: monitor: Add support for logging v2

### PR DESCRIPTION
**Pull request by nordic-krch**
Added support for logging v2 backend API in bluetooth monitor backend.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
@jhedberg I don't know how to run monitor. Can you validate it. I just copy pasted `put` function and adapted it to log2 api changes.

**Comment by jhedberg** [comment]
@nordic-krch there's some information here: https://docs.zephyrproject.org/latest/guides/bluetooth/bluetooth-dev.html?highlight=btmon#embedded-hci-tracing

**Comment by nordic-krch** [comment]
@jhedberg tried it with `nrf52840dk`. I can see that some strings mixed with garbage is printed on the uart but when i start btmon i'm not getting much:
```
btmon --tty /dev/ttyACM0 --tty-speed 115200
Bluetooth monitor ver 5.55
--- /dev/ttyACM0 opened ---
```

**Comment by jhedberg** [comment]
@nordic-krch can you make it work with the original monitor log code? (to determine if this is a setup issue or a bug in your new code)

**Comment by nordic-krch** [comment]
@jhedberg same behavior when checked without this commit.

**Comment by Thalley** [comment]
> @jhedberg same behavior when checked without this commit.

@nordic-krch Which application are you running to test it? Just making sure it's something that actually does something that would generate the log. 

Lately on nRF boards I have had some issues getting the monitor to show anything, perhaps due to the baudrate. Sometimes it would take me quite a few tries with both resetting the board and the `btmon` to get data through.

File: subsys/bluetooth/host/monitor.c
Comment (Thalley) [inline subsys/bluetooth/host/monitor.c:349]: Just making sure this is intentional: `sizeof(id)` would return 3. Did you mean strlen? (same for line 352 and 356).

Comment (nordic-krch) [reply subsys/bluetooth/host/monitor.c:349]: Seems so. it's copied from `monitor_log_put` added by @jhedberg. @jhedberg what do you think?

**Comment by nordic-krch** [comment]
> Which application are you running to test it?

@Thalley I've tried `bluetooth/central`

**Comment by Thalley** [comment]
> > Which application are you running to test it?
> 
> @Thalley I've tried `bluetooth/central`

After connection, that application won't log anything else I suppose. Are you running with a secondary board to trigger a connection? If you want to use that application to test this with, make sure it doesn't actually connect to anything, such that you will continue to see scan reports for verification.

File: subsys/bluetooth/host/monitor.c
Comment (joerchan) [reply subsys/bluetooth/host/monitor.c:349]: @jhedberg Could you take a look?

**Comment by jhedberg** [comment]
Any update with this PR? I really think someone should test that they get btmon output with it before we merge it.

**Comment by Thalley** [comment]
> Any update with this PR? I really think someone should test that they get btmon output with it before we merge it.

@jhedberg Do you have any input on https://github.com/zephyrproject-rtos/zephyr/pull/36921#discussion_r699962826?

File: subsys/bluetooth/host/monitor.c
Comment (jhedberg) [reply subsys/bluetooth/host/monitor.c:349]: Yes, it's intentional since we want the null byte at the end included. `strlen(id) + 1` would give you the same, but that has an added runtime penalty that `sizeof(id)` doesn't have.

**Review by carlescufi — APPROVED** [review]
Tested on nRF52840 with the following config:

```
CONFIG_LOG2_MODE_DEFERRED=y

CONFIG_BT_DEBUG_MONITOR_UART=y
CONFIG_BT_DEBUG_MONITOR_RTT=n

CONFIG_UART_CONSOLE=n
```

**Comment by carlescufi** [comment]
> @jhedberg same behavior when checked without this commit.

@nordic-krch see this: https://github.com/zephyrproject-rtos/zephyr/issues/14753

You need to `nrfjprog -r` for `btmon` to start picking up the data.

**Comment by carlescufi** [comment]
@jhedberg @Thalley @joerchan can we get another approval on this PR?

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
